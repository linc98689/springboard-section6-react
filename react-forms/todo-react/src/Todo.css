.todo-btn{
    margin-left:3px;
    margin-top:3px;
}
.todo-done{
    text-decoration: line-through;
}
.todo-buttons{
    margin-bottom:12px;
}
