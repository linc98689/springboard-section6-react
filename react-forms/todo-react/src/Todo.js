import React from "react";
import "./Todo.css"

const Todo = ({id,task,isdone, removeTask, 
    updateTaskStatus, updateCurrentTaskIdx})=>{
    
    const handleRemove = (evt)=>{
        const idStr = evt.target.id;
        const id = idStr.slice(idStr.indexOf("-")+1);
        removeTask(Number(id));
    };

    const handleDone = (evt) =>{
        const idStr = evt.target.id;
        const id = idStr.slice(idStr.indexOf("-")+1);
        updateTaskStatus(id);
    };

    const handleEdit = (evt) =>{
        const idStr = evt.target.id;
        const id = idStr.slice(idStr.indexOf("-")+1);
        updateCurrentTaskIdx(id);
    };

    // return
    return(
        <li className="todo">
            <span className={isdone==="true"? "todo-done": ""}>{task}</span>
            <div className="todo-buttons">
                <button className="todo-btn" id={`edit-${id}`}
                onClick={handleEdit}>Edit</button>
                <button className="todo-btn" id={`complete-${id}`} 
                isdone={`${isdone}`}
                onClick={handleDone}>Mark as Completed</button>
                <button className="todo-btn" id={`remove-${id}`}
                onClick={handleRemove}>Remove</button>
            </div>
        </li>
    );
}

export default Todo;