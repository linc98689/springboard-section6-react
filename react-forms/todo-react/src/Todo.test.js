import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Todo from "./Todo";

// smoke test
test("should render", ()=>{
    render(<Todo />)
});

// snapshot test
test("should not change", ()=>{
    const {asFragment} = render(<Todo />);
    expect(asFragment()).toMatchSnapshot();
})
