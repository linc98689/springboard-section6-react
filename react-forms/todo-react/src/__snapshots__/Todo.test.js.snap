// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not change 1`] = `
<DocumentFragment>
  <li
    class="todo"
  >
    <span
      class=""
    />
    <div
      class="todo-buttons"
    >
      <button
        class="todo-btn"
        id="edit-undefined"
      >
        Edit
      </button>
      <button
        class="todo-btn"
        id="complete-undefined"
        isdone="undefined"
      >
        Mark as Completed
      </button>
      <button
        class="todo-btn"
        id="remove-undefined"
      >
        Remove
      </button>
    </div>
  </li>
</DocumentFragment>
`;
